engine/code/src/utils/thread.cpp

```diff
@@ -99,12 +99,21 @@ void Thread::Pool::Reload()
     WorkCondition.notify_all();
 
     for (std::thread& thread : Threads) {
-        thread.join();
+        if (thread.joinable()) {
+            thread.join();
+        }
     }
+
+    // Temporary queue to reset all pending tasks of a thread pool
+    std::queue<Task> q;
+    Queue.swap(q);
```

Comment:
This clears the queue of tasks that were left unfinished from the previous scene, and these tasks will become invalid once the resources of the old scene are discarded.